### ISSUE TITLE
Crash when setting up new comparison

### ISSUE THREAD

**Issue by childofsol** [cmmcleod/coriolis#147]
I just started playing around with the comparison tool, making a new comparison of some smuggling ships.

I was refreshing the page after a save so I could add a new ship to the comparison (I am guessing the "add build" list isn't updated in the background when you create a new build in another tab)

Here is the crash I got:

```
Browser:
Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.82 Safari/537.36
Path:
/compare/Smuggling Ships
Error:
unknown
Message:
Cannot read property 'Smuggler' of undefined
Details:
TypeError: Cannot read property 'Smuggler' of undefined
    at r.$scope.addBuild (http://cdn.coriolis.io/app.d7ed0a1f.js:14:1102)
    at http://cdn.coriolis.io/app.d7ed0a1f.js:14:3747
    at Array.forEach (native)
    at new <anonymous> (http://cdn.coriolis.io/app.d7ed0a1f.js:14:3720)
    at e (http://cdn.coriolis.io/lib.df42dcb0.js:6:17209)
    at Object.instantiate (http://cdn.coriolis.io/lib.df42dcb0.js:6:17326)
    at http://cdn.coriolis.io/lib.df42dcb0.js:7:6082
    at http://cdn.coriolis.io/lib.df42dcb0.js:12:23333
    at ea (http://cdn.coriolis.io/lib.df42dcb0.js:7:2331)
    at D (http://cdn.coriolis.io/lib.df42dcb0.js:6:28674)
```

**Comment by childofsol** [comment]
Just got it again - here were my exact steps

- Create new comparison
- Enter name "Smuggler v2" and save
- Add Asp Explorer and Cobra builds. Diamondback Explorer isn't visible. Switched to Diamondback tab, resaved, switched back, refresh, get error

```
Browser:
Mozilla/5.0 (Windows NT 10.0; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/48.0.2564.82 Safari/537.36
Path:
/compare/Smuggling V2
Error:
unknown
Message:
Cannot read property 'Smuggler' of undefined
Details:
TypeError: Cannot read property 'Smuggler' of undefined
    at r.$scope.addBuild (http://cdn.coriolis.io/app.d7ed0a1f.js:14:1102)
    at http://cdn.coriolis.io/app.d7ed0a1f.js:14:3747
    at Array.forEach (native)
    at new <anonymous> (http://cdn.coriolis.io/app.d7ed0a1f.js:14:3720)
    at e (http://cdn.coriolis.io/lib.df42dcb0.js:6:17209)
    at Object.instantiate (http://cdn.coriolis.io/lib.df42dcb0.js:6:17326)
    at http://cdn.coriolis.io/lib.df42dcb0.js:7:6082
    at http://cdn.coriolis.io/lib.df42dcb0.js:12:23333
    at ea (http://cdn.coriolis.io/lib.df42dcb0.js:7:2331)
    at D (http://cdn.coriolis.io/lib.df42dcb0.js:6:28674)
```

**Comment by cmmcleod** [comment]
Can you copy and paste you backup data here: Settings (top right) -> Backup

**Comment by childofsol** [comment]
i've been fiddling since then with the builds so this isn't representative of the state when the crash happened. it started to work, couldn't pin down what changed though

```
{
  "builds": {
    "vulture": {
      "BiWeave BeamPAC Vulture": "24A5A4A3D5A4D3C0v2b0m000206B22bv6662j.Iw19kA==.CwBhGYEYxAmfKVtYqg=="
    },
    "fer_de_lance": {
      "Dream FDL": "26A5A4A4C6A4A3C2c0r0r0r0r0000020l04044a5e32v666.Iw18aQ==.CwBhGYzAmXZkBGZjhqA=",
      "Current FDL": "06B5A4A4D6A4A3C2c1g1g1g1g0006020l04044b2b2bv62j.Iw18aQ==.CwBhGYzAmXZkBGZjhqA=",
      "Hull Tank Plasma Rail FDL": "26A5A4A4A6A4D3C2c2a2a2a2a020204040405B22b2b2766.AwRj4yrI.CwBhGYzAmXYRkUxxVA==",
      "Shield MC FDL": "26A5A4A4D6A4D3C2c272727270404040404044a5e5e2766.AwRj4yNMg===.CwBhGYzAmXYRkUxxVA==",
      "Next FDL": "06A5A4A4C6A4A3C2c1a1a1a1a0005020l04044c2b2bv62j.AwRj4yrI.CwBhGYzAmXZkBGZjhqA=",
      "Target FDL": "06A5A4A4D6A4A3C2c1g1g1g1g000l020604044b2b2bv62j.Iw18aQ==.CwBhGYzAmXZkBGZjhqA="
    },
    "viper_mk_iv": {
      "Viper Mk4 Railguns": "24A4A4A2D3A3D4C2a2a1818020mB32b2t27v669.Iw1-kA==.Aw1-kA=="
    },
    "viper": {
      "Stealth Viper": "03A3A3A2D3A3D2C27272424020229293r69.AwRj4yyA.CwBhGYEYxAmfaWiSwg=="
    },
    "asp": {
      "Asp Long Range Smuggler": "04A5A5A4D4D5D5C----------4d39--F9v62f.Iw18WQ==.Aw18WQ=="
    },
    "diamondback_explorer": {
      "Smuggler": "04A4E5E3E4E3E5C-1717----490201-2h.Iw1-kA==.Aw1-kA=="
    },
    "cobra_mk_iii": {
      "Smuggler": "04A4A4A3A3A3D4C2a2a2929000232034501v62f.Iw1-kA==.CwBhGYzAmXrUBGEog==="
    }
  },
  "comparisons": {
    "Smuggling Ships": {
      "facets": [
        9,
        6,
        4,
        1,
        3,
        2
      ],
      "builds": [
        {
          "shipId": "asp",
          "buildName": "Asp Long Range Smuggler"
        },
        {
          "shipId": "diamondback_explorer",
          "buildName": "Smuggler"
        }
      ]
    },
    "Smuggling V2": {
      "facets": [
        9,
        6,
        4,
        1,
        3,
        2
      ],
      "builds": [
        {
          "shipId": "asp",
          "buildName": "Asp Long Range Smuggler"
        },
        {
          "shipId": "cobra_mk_iii",
          "buildName": "Smuggler"
        }
      ]
    }
  },
  "insurance": null,
  "discounts": null
}
```

**Comment by cmmcleod** [comment]
Ok, I was able to replicate this. The error occurs when localstorage and state get out of sync (when running multiple tabs only). Specifically when all builds for a specific ship have been deleted in one tab, and they are included in comparisons that are open in a second tab.

This is already a non-issue in 2.0 (currently in alpha stage).

Thanks for the report!

**Comment by childofsol** [comment]
No problem!

**Comment by cmmcleod** [comment]
Closing for now. Confirmed issue is fixed in 2.0
